gcc/jit/jit-recording.cc

```diff
@@ -5188,6 +5188,21 @@ recording::block::get_loc () const
     return NULL;
 }
 
+/* Set the source-location of a block by using that of the first
+   statement within it, if any.  */
+
+void
+recording::block::set_loc (location * loc)
```

Comment:
Does it have any effect on the generated code to set the location of a block like this? I would think that setting the locations on the statements directly should be enough, no?

Reply:
I was actually uncertain about whether to add such a `set_loc` for gccjit or to simply allow setting locations for all statements if possible. When the code was written, I was unsure about the future.
In the current version only rvalue is used. Maybe I should simply remove all other code.

Reply:
> Maybe I should simply remove all other code.

I guess it depends if you're sure we'll use all the methods you added.

Reply:
No. Even in the long round for set location of a block.

But there is a chance where I or someone else will use the other remaining set locations, except the rvalue one in use.

In that case, should I keep everything else not currently in use or simply commit ONLY the `rvalue::set_loc`?

Reply:
Maybe, if someone plans to separate debuginfo from code generation, as is done in llvm.
But of course, we can still leave it as it is, as I explained in my cg_gcc commit.

Reply:
I'm still not sure I understand after reading your comment [here](https://github.com/rust-lang/rustc_codegen_gcc/pull/455#issuecomment-1962828082), so let me know if this is correct: these methods will be used when rustc_codegen_gcc will not set the debuginfo in `builder.rs`, but will set them separately like is done for `rustc_codegen_llvm`?

Reply:
I'm still not sure I understand after reading your comment [here](https://github.com/rust-lang/rustc_codegen_gcc/pull/455#issuecomment-1962828082), so let me know if this is correct: these methods will be used when rustc_codegen_gcc will not set the debuginfo in builder.rs, but will set them separately like is done for rustc_codegen_llvm?

Reply:
Yes.